Error on installing Helm chart with nodeSelector value

It is impossible to install Helm chart with nodeSelector value, the error message is:


```
# helm template rqlite rqlite/rqlite --namespace rqlite --version 1.12.0 -f custom-values.yaml                 
coalesce.go:286: warning: cannot overwrite table with non table for rqlite.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution (map[nodeSelectorTerms:[map[matchExpressions:[map[key:kubernetes.io/arch operator:In values:[amd64 386 arm arm64]]]]]])
Error: YAML parse error on rqlite/templates/statefulset.yaml: error converting YAML to JSON: yaml: line 114: did not find expected '-' indicator

Use --debug flag to render out invalid YAML
```

The value specified in the custom-values.yaml is

```
nodeSelector:
  kubernetes.io/arch: "arm64"
```


And in the statefulset.yaml it appears within container's arguments, here is an example

```

---
# Source: rqlite/templates/statefulset.yaml

apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: rqlite
  labels:
    helm.sh/chart: rqlite-1.12.0
    app.kubernetes.io/name: rqlite
    app.kubernetes.io/instance: rqlite
    app.kubernetes.io/version: "8.30.5"
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/component: voter
spec:
  replicas: 2
  # rqlite is tolerant of all nodes coming up simultaneously, so we set the pod management
  # policy to parallel to allow fast scaling.
  podManagementPolicy: Parallel
  updateStrategy:
    type: RollingUpdate
  selector:
    matchLabels:
      app.kubernetes.io/name: rqlite
      app.kubernetes.io/instance: rqlite
      app.kubernetes.io/component: voter
  serviceName: rqlite-headless

  template:
    metadata:
      labels:
        app.kubernetes.io/name: rqlite
        app.kubernetes.io/instance: rqlite
        app.kubernetes.io/component: voter
    spec:
      automountServiceAccountToken: false

      terminationGracePeriodSeconds: 15
      securityContext:
        fsGroup: 10050
        fsGroupChangePolicy: OnRootMismatch
        runAsGroup: 10050
        runAsUser: 10050
      nodeSelector:
        kubernetes.io/arch: arm64
      tolerations:
        - effect: NoSchedule
          key: cpu
          operator: Equal
          value: ARM
      affinity:
        podAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
          - labelSelector:
              matchExpressions:
              - key: app.kubernetes.io/name
                operator: In
                values:
                - rqlite
            topologyKey: kubernetes.io/hostname

      volumes:
        - name: secrets
          secret:
            secretName: rqlite
            defaultMode: 288 # 0400
        - name: extra
          secret:
            secretName: rqlite-extra
            defaultMode: 288 # 0400
        - name: config-peers
          configMap:
            name: rqlite-peers
      containers:
        - name: rqlite
          image: docker.io/rqlite/rqlite:8.30.5
          # Override the default docker entrypoint script so we can add our own
          # custom startup logic ahead of rqlited. Afterward we exec the original
          # entrypoint script with all the required arguments.
          command: ["/bin/sh", "-c"]
          args:
            - |-
              # If we're using static peers, copy the chart-generated peers file into
              # place. But first remove any existing peers file, which we do regardless as
              # in normal operation we use DNS discovery.
              rm -f /rqlite/raft/peers.info
              if [ "$RQLITE_USE_STATIC_PEERS" = "true" ]; then
                echo "WARNING: Using generated static peers. This is a recovery procedure and must be reverted after service is restored."
                cat /config/peers/peers.json
                mkdir -p /rqlite/raft
                cp /config/peers/peers.json /rqlite/raft
              fi
              # Exec original entrypoint and pass all arguments through, including $0
              # which is actually the first argument due to this inline script being
              # executed as a command string.
              exec /bin/docker-entrypoint.sh "$0" "$@"

            # All arguments after this point are passed through to docker-entrypoint.sh
            # by the init script above.
            - -disco-mode=dns-srv
            - -disco-config={"name":"rqlite-headless","service":"raft"}
            # This is value is interpolated by the Kubelet using the Kubernetes
            # substitution syntax instead of being rendered by the Helm chart to ensure
            # that changes to replicaCount doesn't result in a rolling restart of all pods
            # in the StatefulSet.
            - -bootstrap-expect=$(RQLITE_VOTER_REPLICA_COUNT)
            - -join-interval=1s
            - -join-attempts=120
            - -raft-shutdown-stepdown
           kubernetes.io/arch: arm64
            - -fk=true
          resources:
            requests:
              cpu: 100m
              memory: 128Mi
 ...
```

--- a/charts/rqlite/templates/_statefulset.tpl
+++ b/charts/rqlite/templates/_statefulset.tpl
@@ -219,9 +219,6 @@ spec:
             - -join-interval=1s
             - -join-attempts=120
             - -raft-shutdown-stepdown
-            {{- with dig "nodeSelector" $.Values.nodeSelector $values }}
-              {{- tpl (toYaml .) $ | nindent 12 }}
-            {{- end }}
             {{- with $.Values.extraArgs }}
               {{- tpl (toYaml .) $ | nindent 12 }}
             {{- end }}

